### ISSUE TITLE
The picture shows it upside down

### ISSUE THREAD

**Issue by DevWhan** [SDWebImage/SDWebImage#3594]
Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 16.6
 SDWebImage Version      | 5.18.0
 Integration Method      | cocoapods
 Xcode Version           | Version 14.2 (14C18)
 Repro rate              | this image url all the time (100%), other sometimes 
 Repro with our demo prj | yes, but can't give out the url
 Demo project link       | https://github.com/DevWhan/IssuesSDWebImage

The same url appears fine in the browser, but the sd image is upside down
The screenshot is in the image directory of demo

**Comment by dreampiggy** [comment]
If you don't provide the image url, at least you must provide a local image which can reproduce your problem.

SDWebImage's decoder supports XMP metadata with image orientation, I think this is because you use some transfomer or advanced feature which does not handle the orientation at all.

**Comment by DevWhan** [comment]
> If you don't provide the image url, at least you must provide a local image which can reproduce your problem.
> 
> Current Demo you provided does not work at all...
> 
> SDWebImage's decoder supports XMP metadata with image orientation, I think this is because you use some transfomer or advanced feature which does not handle the orientation at all.

I've already provided the url in the demo

In addition, I found that loading this address in Safari will occasionally reverse the problem, more refresh will appear a few times, and in iOS is required to appear, now I do not know whether this is related to SD, please help me to look at it, thank you very much
Environment: macOS Ventura Safari 16.5.2 (18615.2.9.11.10)

**Comment by dreampiggy** [comment]
> Printing description of properties:
{
    ColorModel = RGB;
    DPIHeight = 300;
    DPIWidth = 300;
    Depth = 8;
    Orientation = 3;
    PixelHeight = 3024;
    PixelWidth = 4032;
    ProfileName = "Display P3";
    "{JFIF}" =     {
        DensityUnit = 1;
        JFIFVersion =         (
            1,
            0,
            1
        );
        XDensity = 300;
        YDensity = 300;
    };
    "{TIFF}" =     {
        Orientation = 3;
    };
}
(lldb)

**Comment by dreampiggy** [comment]
The behavior does not match Apple's documentation, so it's Apple's bug.

I do not have a good solution currently.

**Comment by dreampiggy** [comment]
For your `normal image`, the properties shows:

```
{
    ColorModel = RGB;
    Depth = 8;
    Orientation = 3;
    PixelHeight = 3024;
    PixelWidth = 4032;
    PrimaryImage = 1;
    ProfileName = "Display P3";
    "{TIFF}" =     {
        Orientation = 3;
    };
}
```

**Comment by dreampiggy** [comment]
Apple's `[UIImage imageWithData:]` pass a private options here:

```
po $arg3
{
    kCGImageSourceSkipMetadata = 1;
}
```

**Comment by dreampiggy** [comment]
When use `kCGImageSourceSkipMetadata = 1`, the returned metadata shows the correct value:

```
(lldb) po properties
{
    ColorModel = RGB;
    DPIHeight = 300;
    DPIWidth = 300;
    Depth = 8;
    PixelHeight = 3024;
    PixelWidth = 4032;
    ProfileName = "Display P3";
    "{JFIF}" =     {
        DensityUnit = 1;
        JFIFVersion =         (
            1,
            0,
            1
        );
        XDensity = 300;
        YDensity = 300;
    };
}
```

**Comment by dreampiggy** [comment]
A quick workaround is to use this SPI `kCGImageSourceSkipMetadata`, like Apple's own WebKit engine :(

https://github.com/adobe/webkit/blob/067e0a8adb3c27c682d894072098c405fb5ca386/Source/WebCore/platform/graphics/cg/ImageSourceCG.cpp#L46

Really suck

**Comment by dreampiggy** [comment]
Fixed in 5.8.1

**Comment by therehello** [comment]
@dreampiggy 
This issue appears to persist.

SDWebImage Version latest

<img width="1778" height="1862" alt="Image" src="https://github.com/user-attachments/assets/7cb4d357-338c-4d59-b0bd-82bb24059319" />
